src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,106 @@
-import React from 'react';
-import { NewCommentForm } from '../NewCommentForm';
+import { FC, useEffect, useState } from 'react';
+
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+import { NewCommentForm } from '../NewCommentForm';
+import { getPostComments, postComment, removeComment } from '../../api/comments';
+import { getPostDetails } from '../../api/posts';
+import { Loader } from '../Loader';
+
+type Props = {
+  postId: number,
+};
+
+export const PostDetails: FC<Props> = ({ postId }) => {
+  const [details, setDetails] = useState<Post | null>(null);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [commentsVisibility, setCommentsVisibility] = useState<boolean>(true);
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  useEffect(() => {
+    getPostDetails(postId)
+      .then(postDetails => {
+        setDetails(postDetails);
+        setIsLoaded(true);
+      });
+
+    getPostComments(postId)
+      .then(postComments => setComments([...postComments]));
+  }, [postId]);
+
+  const onDelete = (commentId: number) => {
+    removeComment(commentId)
+      .then(response => {
+        if (response) {
+          setComments(prevComments => prevComments
+            .filter(prevComment => prevComment.id !== commentId));
+        }
+      });
+  };
+
+  const onAdd = (newComment: Partial<Comment>) => {
+    postComment(newComment)
+      .then(response => setComments([...comments, response]));
+  };
+
+  if (!isLoaded) {
+    return (<Loader />);
+  }
+
+  return (
+    <div className="PostDetails">
+      <h2>
+        {`Post details: ${comments.length}`}
```

Comment:
That's unnecessary, I can see comments count on hide/show button
```suggestion
        Post details:
```